sites/docs/src/content/docs/tutorials/tests_and_test_data/components/03_testing_modules.md

```diff
@@ -0,0 +1,810 @@
+- `when` - Test input setup and execution parameters
+- `then` - Assertions to verify expected behavior and outputs
+
+:::note
+**Key points:**
+
+- Script paths starting with `./` or `../` are relative to the test script's location.
+  :::
+
+## Understanding snapshots
+
+**What are snapshots?**
+
+Snapshots are nf-test's way of capturing and validating the expected outputs of your tests. When you run a test for the first time, nf-test creates a `.snap` file containing checksums, file names, and other metadata about the outputs.
+
+**How snapshot matching works:**
+
+1. **First run**: nf-test generates a snapshot file (e.g., `main.nf.test.snap`) containing the expected outputs
+2. **Subsequent runs**: nf-test compares current outputs against the stored snapshot
+3. **Pass/Fail**: Test passes if outputs match the snapshot, fails if they differ
+
+**Why snapshots are useful:**
+
+- **Automated validation**: No need to manually check every output file
+- **Regression detection**: Automatically catch when module behavior changes
+- **Comprehensive checking**: Validates file content, structure, and metadata
+
+**Example snapshot content:**
+
+```json
+{
+  "sarscov2 - fastq": {
+    "content": [
+      {
+        "0": ["versions.yml:md5,c50aa59475ab20752531545a585f8f2d"],
+        "1": [
+          [
+            {
+              "id": "test",
+              "single_end": false
+            },
+            "test.fastq.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
+          ]
+        ]
+      }
+    ]
+  }
+}
+```
+
+## Your first simple test walkthrough with an example
+
+Let's walk through creating a basic test step-by-step using the `cat` module as an example:
+
+### Step 1: Examine the module
+
+First, let's look at a simple module (`modules/cat/main.nf`):
+
+```groovy
+process CAT {
+    input:
+    tuple val(meta), path(file_in)
+
+    output:
+    tuple val(meta), path("${prefix}"), emit: file_out
+    path "versions.yml"               , emit: versions
+
+    script:
+    prefix = "${meta.id}.txt"
+    """
+    cat ${file_in} > ${prefix}
+    """
+}
+```
+
+This process simply concatenates input files and outputs a single file.
+
+### Step 2: Write the test
+
+Create `modules/cat/tests/main.nf.test`:
+
+```groovy
+nextflow_process {
+    name "Test Process CAT"
+    script "../main.nf"
+    process "CAT"
+
+    test("simple file concatenation") {
+        when {
+            process {
+                """
+                input[0] = [
+                    [ id:'test' ], // meta map with sample identifier
+                    file('test_file.txt')  // input file
+                ]
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },           // Check process completed successfully
+                { assert snapshot(process.out).match() } // Validate outputs match expected snapshot
+            )
+        }
+    }
+}
+```
+
+**Key components explained:**
+
```

Comment:
I would  actually describe the top level name/script/process/`when/test` scopes are for  and then finally  move into the specific detail of the id/file/process.success/snapshot